tests/pages/contribute/signup.py

```diff
@@ -0,0 +1,115 @@
+    _message_locator = (By.ID, 'person-message')
+    _html_format_locator = (By.ID, 'id_format_0')
+    _text_format_locator = (By.ID, 'id_format_1')
+    _privacy_policy_checkbox_locator = (By.ID, 'id_privacy')
+    _newsletter_checkbox_locator = (By.ID, 'id_newsletter')
+    _submit_button_locator = (By.CSS_SELECTOR, '.submit > button')
+    _success_url_slug = '/contribute/thankyou/'
+
+    @property
+    def category_areas(self):
+        return [CategoryArea(self.selenium, root=el) for el in
+                   self.selenium.find_elements(*self._category_areas_locator)]
+
+    @property
+    def contributor_categories(self):
+        return [ContributorCategory(self.selenium, root=el) for el in
+                   self.selenium.find_elements(*self._contributor_categories_locator)]
+
+    def type_name(self, value):
+        self.selenium.find_element(*self._name_locator).send_keys(value)
+
+    def type_email(self, value):
+        self.selenium.find_element(*self._email_locator).send_keys(value)
+
+    def select_country(self, value):
+        el = self.selenium.find_element(*self._country_locator)
+        Select(el).select_by_visible_text(value)
+
+    def select_html_format(self):
+        self.selenium.find_element(*self._html_format_locator).click()
+
+    def select_text_format(self):
+        self.selenium.find_element(*self._text_format_locator).click()
+
+    def accept_privacy_policy(self):
+        el = self.selenium.find_element(*self._privacy_policy_checkbox_locator)
+        assert not el.is_selected(), 'Privacy policy has already been accepted'
+        el.click()
+        assert el.is_selected(), 'Privacy policy has not been accepted'
+
+    @property
+    def is_areas_field_displayed(self):
+        return self.is_element_displayed(self._areas_locator)
+
+    def click_start_contributing(self):
+        self.selenium.find_element(*self._submit_button_locator).click()
+        Wait(self.selenium, self.timeout).until(lambda s: self.sign_up_successful)
+
+    @property
+    def sign_up_successful(self):
+        return self._success_url_slug in self.selenium.current_url
+
+
+class ContributorCategory(PageRegion):
+
+    _category_locator = (By.TAG_NAME, 'label')
+
+    @property
+    def select_category(self):
+        self.root.find_element(*self._category_locator).click()
+
+
+class CategoryArea(PageRegion):
+
+    _area_locator = (By.TAG_NAME, 'select')
+
+    def select_area(self, value):
+        el = self.selenium.find_element(*self._area_locator)
+        Select(el).select_by_visible_text(value)
+
+    @property
+    def changes_to_required_field(self):
+        assert not self.is_required, 'Area field is already required'
+        Wait(self.selenium, self.timeout).until(lambda s: self.is_required)
+        return self.is_required
+
+    @property
+    def is_required(self):
+        el = self.selenium.find_element(*self._area_locator)
+        return el.get_attribute('required') == 'true'
```

Comment:
The idea here is that the areas are only set as required when they are made visible, hence the need to check for these cases specifically.